src/QueryPipeline/QueryPipelineBuilder.cpp

```diff
@@ -393,6 +398,17 @@ std::unique_ptr<QueryPipelineBuilder> QueryPipelineBuilder::joinPipelines(
         left->pipe.processors.emplace_back(std::move(joining));
     }
 
+    /// Move the collected processors to the last step in the right pipeline.
+    Processors processors = collector.detachProcessors();
+    if (auto * expression_step = typeid_cast<ExpressionStep*>(step))
```

Comment:
is this special to `ExpressionStep` or we can do it for any `ITransformingStep`?

Reply:
Seems the top step of the right pipeline is `ExpressionStep` right now, so I think it's ok to specify the type explicitly and still use `collected_processors `for other unexplicitly types at the same time.

Reply:
you distribute the steps in the pipeline description the right way: `JoinTransform` is over two pipelines and `FillingRightJoinSide` is clearly relates to the right pipeline. and because of how `joinPipelines` is written now I see no reason to add `processors` to the `collected_processors`.

Reply:
done